lib/vaahextendflutter/services/storage/local/hive/hive_storage_impl.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:hive/hive.dart';
+
+import '../../storage.dart';
+
+/// A class implementing Storage interface using Hive as storage backend.
+class HiveStorageImpl implements Storage {
+  final String name;
+
+  Box? _box;
+
+  HiveStorageImpl({required this.name});
```

Comment:
make this not required with default value "VaahFlutterHiveBox" or something like that